server-common/src/test/java/org/apache/kafka/server/common/MetadataVersionTest.java

```diff
@@ -72,20 +71,18 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 class MetadataVersionTest {
-
     @Test
     public void testFeatureLevel() {
-        MetadataVersion[] metadataVersions = MetadataVersion.VERSIONS;
-        int firstFeatureLevelIndex = Arrays.asList(metadataVersions).indexOf(MetadataVersion.MINIMUM_KRAFT_VERSION);
-        for (int i = 0; i < firstFeatureLevelIndex; i++) {
-            assertTrue(metadataVersions[i].featureLevel() < 0);
+        int i = 0;
+        while (i < MetadataVersion.VERSIONS.length &&
```

Comment:
nit: The new version of this test seems more obscure than the original. It seems more straightforward to assert 1) all versions below IBP_3_0_IV1 have -1 feature level, and 2) that all subsequent version increment feature level by 1. Also, it would be helpful to have a better name for the test so that we don't have to decipher what it's trying to do. For example, perhaps `testSequentialFeatureLevel` or something like that.

Reply:
Good points. I will separate this out into `testKRaftFeatureLevelsBefore3_0_IV1` and `testKRaftFeatureLevelsAtAndAfter3_0_IV1`, which are hopefully self-explanatory.